src/rules/noNullUndefinedUnionRule.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * @license
+ * Copyright 2019 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    isParameterDeclaration,
+    isPropertyDeclaration,
+    isPropertySignature,
+    isSignatureDeclaration,
+    isTypeAliasDeclaration,
+    isTypeReference,
+    isUnionType,
+    isVariableDeclaration,
+} from "tsutils";
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.TypedRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "no-null-undefined-union",
+        description: "Disallows union types with both `null` and `undefined` as members.",
+        rationale: Lint.Utils.dedent`
+            A union type that includes both \`null\` and \`undefined\` is either redundant or fragile.
+            Enforcing the choice between the two allows the \`triple-equals\` rule to exist without
+            exceptions, and is essentially a more flexible version of the \`no-null-keyword\` rule.
+        `,
+        optionsDescription: "True if the rule should be enabled.",
```

Comment:
There are no options. You should use:

```suggestion
        optionsDescription: "Not configurable.",
```